plugins/trilinos/CMakeLists.txt

```diff
@@ -0,0 +1,146 @@
+MESSAGE("   Trilinos_TPL_LIST = ${Trilinos_TPL_LIST}")
+MESSAGE("   Trilinos_TPL_INCLUDE_DIRS = ${Trilinos_TPL_INCLUDE_DIRS}")
+MESSAGE("   Trilinos_TPL_LIBRARIES = ${Trilinos_TPL_LIBRARIES}")
+MESSAGE("   Trilinos_TPL_LIBRARY_DIRS = ${Trilinos_TPL_LIBRARY_DIRS}")
+MESSAGE("   Trilinos_BUILD_SHARED_LIBS = ${Trilinos_BUILD_SHARED_LIBS}")
+MESSAGE("End of Trilinos details\n")
+
+# Make sure to use same compilers and flags as Trilinos
+#MESSAGE("   CMAKE_CXX_COMPILER = ${CMAKE_CXX_COMPILER}")
+#MESSAGE("   Trilinos_CXX_COMPILER = ${Trilinos_CXX_COMPILER}")
+#MESSAGE("   CMAKE_CXX_FLAGS = ${CMAKE_CXX_FLAGS}")
+#MESSAGE("   Trilinos_CXX_COMPILER_FLAGS =  ${Trilinos_CXX_COMPILER_FLAGS} ")
+
+#set(CMAKE_CXX_COMPILER ${Trilinos_CXX_COMPILER} )
+set(CMAKE_CXX_FLAGS  "${Trilinos_CXX_COMPILER_FLAGS} ${CMAKE_CXX_FLAGS}")
+
+
+# ----------------------------------------------------------------------------
+# ----------------------------------------------------------------------------
+
+add_library(trilinos_wrapper
+        include/alien/trilinos/backend.h
+        include/alien/trilinos/options.h
+        include/alien/trilinos/trilinos_config.h
+        src/trilinos_vector.cpp
+        src/trilinos_vector.h
+        src/trilinos_matrix.cpp
+        src/trilinos_matrix.h
+        src/trilinos_linear_algebra.cpp
+        src/trilinos_linear_solver.cpp
+        src/trilinos_linear_solver.h
+        src/converters/trilinos_to_simplecsr_vector.cpp
+        src/converters/simplecsr_to_trilinos_vector.cpp
+        src/converters/simplecsr_to_trilinos_matrix.cpp
+        src/trilinos_config.cpp
+        )
+
+target_link_libraries(trilinos_wrapper PUBLIC
+        Alien::alien_core
+        )
+
+target_link_libraries(trilinos_wrapper PRIVATE
+        MPI::MPI_CXX
+        Arccore::arccore_trace
+        Arccore::arccore_collections
+        Arccore::arccore_base
+        Arccore::arccore_message_passing_mpi
+        ${Trilinos_LIBRARIES}
+        ${Trilinos_TPL_LIBRARIES}
+        )
+
+if (ALIEN_PLUGIN_TRILINOS_OMP)
+    message("------ COMPILING TRILINOS WRAPPER WITH *OPENMP* BACKEND  ------- ")
+elseif (ALIEN_PLUGIN_TRILINOS_CUDA)
+    message("------ COMPILING TRILINOS WRAPPER WITH *CUDA* BACKEND  ------- ")
+elseif (ALIEN_PLUGIN_TRILINOS_HIP)
+    message("------ COMPILING TRILINOS WRAPPER WITH *HIP* BACKEND  ------- ")
+elseif (ALIEN_PLUGIN_TRILINOS_SERIAL)
+    message("------ COMPILING TRILINOS WRAPPER WITH *SERIAL* BACKEND  ------- ")
+elseif (ALIEN_PLUGIN_TRILINOS_THREADS)
+    message("------ COMPILING TRILINOS WRAPPER WITH *PTHREAD* BACKEND  ------- ")
+elseif (ALIEN_PLUGIN_TRILINOS_SYCL)
+    message("------ COMPILING TRILINOS WRAPPER WITH *SYCL* BACKEND  ------- ")
+else ()
+    message(FATAL_ERROR "------ NO HARDWARE EXECUTOR SPECIFIED FOR TRILINOS WRAPPER  ------- ")
+endif ()
+
+configure_file(
+        Trilinos.config.in
+        ${CMAKE_CURRENT_BINARY_DIR}/include/alien/ginkgo/TrilinosConfig.h
```

Comment:
right -> renamed to Trilinos.backend.in and TrilinosBackend.h
not sure they should be merged : 
- the backend file specifies the Kokkos node, and the header is generated by the cmake
- the config file specifies the types used, and is not a generated file
btw, the compile time backend choice should disappear in the future, since we aim to switch at runtime.